src/libbson/tests/test-string.c

```diff
@@ -356,9 +357,70 @@ test_bson_string_capacity (void *unused)
       large_str[UINT32_MAX - 2u] = 's'; // Restore.
    }
 
+   // Test allocating the largest possible string.
+   {
+      bson_string_t *str = _bson_string_alloc (UINT32_MAX - 4u);
+      ASSERT_CMPUINT32 (str->alloc, ==, UINT32_MAX);
+      ASSERT_CMPUINT32 (str->len, ==, 0);
+      bson_string_free (str, true);
+   }
+
    bson_free (large_str);
 }
 
+static void
+test_bson_string_append_ex (void)
+{
+   bson_string_t *str;
+   char *s;
+
+   str = bson_string_new (NULL);
+   _bson_string_append_ex (str, "the quick brown fox jumps over the lazy dog", 10);
+   s = bson_string_free (str, false);
+   ASSERT_CMPSTR (s, "the quick ");
+   bson_free (s);
```

Comment:
```suggestion
   s = bson_string_free (str, false);
   ASSERT_CMPSTR (str->str, "the quick ");
   bson_string_free (str, true);
```
Simplification. Local variable `s` is not needed.